### ISSUE TITLE
MAP -> ODOM  the odom position is changing in respect to the map  when a robot is moving

### ISSUE THREAD

**Issue by lmendyk** [ros-planning/navigation2#4135]
I’m testing turtlebot3_manipulation_navigation. Initially with SLAM, but observed strange behavior so finally I reduced the test case to just the navigating with no SLAM.
First I use the button to define the initial robot position.  After that I can see correctly the map, odom and base_footprint frames presented in rviz2. 

But when I command the robot to do the twist – the robot actually makes nice circles  on the floor,  but in rviz – I can see that the odom frame is also rotating and the whole robot is making strange movements on top of the map.  After changing the fixed frame from map to odom, I can see that the base_footprint is making nice circles as the robot is actually doing. But to my surprise the map frame is also making the circles following the base-footprint frame (it lags slightly behind).

To my understanding the Odom frame should be fixed in respect to map, and when the robot moves only the base-footprint should change in respect to odom. 

I don’t understand what’s happening, don’t know what I need to check. 

This behavior makes SLAM and then navigation impossible to produce good results. 

I’m using  humble version of navigation stack.

**Comment by SteveMacenski** [comment]
ask on the TB3 demo you're attempting, I can't speak to every open-source project's proper use of this library and the other tools in the ecosystem :wink:

**Comment by lmendyk** [comment]
OK, I think I have found the problem.   

The lidar on top of the robot has moved from the center of the robot to the right so when the robot is making the circles the the physicals center of the lidar must be doing cielce  around it  theoretical center (that's my interpretation) .  
 Once I have corrected its physical position the lidar, the problem seems to be solved - I will have to do more test to be completely sure.

So I does not look like specific to open-source project problem but proves the importance of URDF matching the physical construction of the robot.
